### ISSUE TITLE
Reconnect on disconnect panics with close of closed channel

### ISSUE THREAD

**Issue by ajmckee** [wagslane/go-rabbitmq#5]
**RabbitMq Version**: 3.8..9
**GoLang Version:** go1.15.3 darwin/amd64
**Reproduce:**
Using the [example consumer](https://github.com/wagslane/go-rabbitmq/blob/main/examples/consumer/main.go) and RabbitMq 3.8.9  within a docker container, a forced restart of rabbitmq initiates the channelManager reconnection strategy commences. However  on reconnection to the rabbitmq service the channel is closed and a panic issued with "close of a closed channel"

**Desired Behaviour** Reconnection and rebinding of consumer handlers
```
2021/04/23 07:13:12 gorabbit: waiting 8s seconds to attempt to reconnect to amqp server
2021/04/23 07:13:20 gorabbit: successfully reconnected to amqp server after close
2021/04/23 07:13:20 gorabbit: successfully reconnected to amqp server after close
panic: close of closed channel

goroutine 36 [running]:
github.com/wagslane/go-rabbitmq.(*channelManager).startNotifyCancelOrClosed(0xc000140600)
        /Users/ajmckee/go/pkg/mod/github.com/wagslane/go-rabbitmq@v0.5.0/channel.go:69 +0x294
created by github.com/wagslane/go-rabbitmq.newChannelManager
        /Users/ajmckee/go/pkg/mod/github.com/wagslane/go-rabbitmq@v0.5.0/channel.go:32 +0x167
panic: close of closed channel```

**Comment by wagslane** [comment]
Should be fixed in 0.5.1, let me know if you see more issues, thanks!

**Comment by ajmckee** [comment]
@wagslane Yep, works a treat now. Thank you!!
